### ISSUE TITLE
Okta Login does not support provider flexibility during Sync

### ISSUE THREAD

**Issue by eliatcodecov** [codecov/engineering-team#910]
# Problem
After login via okta, the user is redirected to a /sync page to connect their SCM. This page does not consider self-hosted providers. For example a "Login to Github Enterprise" button is not present. So if an install of Codecov is using Okta _and_ self-hosted Github Enterprise, they cannot complete their okta login. 

# Solution
The /sync page should likely build its login button / behavior similar to how the login page of enterprise works today. On an enterprise install, this page is aware of the providers in the codecov.yml and uses that provider information to build various login/oauth buttons. 

(Note, this was initially brought up as a [feedback discussion issue](https://github.com/codecov/feedback/discussions/176) ) 

## Blocked by
- https://github.com/codecov/engineering-team/issues/947

**Comment by nicholas-codecov** [comment]
Will need to create another ticket for some API changes that will be required for to accomplish this

**Comment by nicholas-codecov** [comment]
Created new ticket #947 for required API changes.

cc @katia-sentry
